src/mgmt/cli_mgmt.c

```diff
@@ -697,6 +718,20 @@ dc_mgmt_net_cfg(const char *name, crt_init_options_t *crt_info)
 		"\tD_PROVIDER: %s, CRT_TIMEOUT: %d, CRT_SECONDARY_PROVIDER: %s\n",
 		crt_info->cio_provider, crt_info->cio_crt_timeout, buf);
 
+	/* Save attached ranks id info */
+	g_num_serv_ranks = resp->n_rank_uris;
+	serv_ranks_tmp   = NULL;
+	if (g_num_serv_ranks > 0) {
+		D_ALLOC_ARRAY(serv_ranks_tmp, g_num_serv_ranks);
+		if (serv_ranks_tmp == NULL)
+			D_GOTO(cleanup, rc = -DER_NOMEM);
+		for (idx = 0; idx < g_num_serv_ranks; idx++)
+			serv_ranks_tmp[idx] = resp->rank_uris[idx]->rank;
+	}
```

Comment:
any reason for not using D_REALLOC_ARRAY instead of tmp array?

Reply:
As we do not have concurrency issues it is not needed, however I prefer to always have a consistent value in the `g_serv_ranks` environment variable.  I also do not think that we could have performance constraints on this portion of code and thus `D_REALLOC_ARRAY` seems to not be mandatory.